src/internals.jl

```diff
@@ -103,8 +102,8 @@ Base.getindex(c::ViewChunks{T,C,S}, i::Int) where {T,C,S} = @view(c.collection[g
 
 Base.eltype(::IndexChunks{T,C,Consecutive}) where {T,C} = UnitRange{Int}
 Base.eltype(::IndexChunks{T,C,RoundRobin}) where {T,C} = StepRange{Int,Int}
-Base.eltype(c::ViewChunks{T,C,Consecutive}) where {T,C} = typeof(c[firstindex(c)])
-Base.eltype(c::ViewChunks{T,C,RoundRobin}) where {T,C} = typeof(c[firstindex(c)])
+Base.eltype(c::ViewChunks{T,C,Consecutive}) where {T,C} = typeof(@view(c.collection[firstindex(c.collection):lastindex(c.collection)]))
```

Comment:
These changes are important to allow `eltype` to return the proper types without having to call `getchunkindices`. `firstindex` and `lastindex` are considered part of the required interface.